### PR TITLE
doc (webapi): jekyll / paradox based documentation website

### LINKED ISSUE #733: Thinking about better (read: nicer looking, easier to use, etc.) documentation
There is now an extension to markdown, allowing us to do all the things that made us change to RST in the first place:

- https://github.com/lightbend/paradox
- https://developer.lightbend.com/docs/paradox/latest/

Main advantages:
- based on markdown and thus easier to use than RST. we could simply copy/paste markdown from issues/pull-request. think of release notes, documentation, examples.
- source (markdown) also directly usable on Github (links between pages work). no need to compile anything to have the newest documentation
- making it easier to use, lowers the barrier for developers to actually write documentation (which is enough painful in itself)
- developed and maintained by the Akka people

Disadvantages:
- need to convert RST to MD but this could be automated with a [script](https://gist.github.com/zaiste/77a946bbba73f5c4d33f3106a494e6cd) using pandoc.

### LINKED ISSUE #734: Thinking about better www.knora.org website
Github pages support https://jekyllrb.com. Maybe we could create a simple site with some nice styling using jekyll. All pages would be written in markdown and the automatically build and published by Github.

### PR DESCRIPTION
**Pull request by subotic**
### Summary

- [x] add sbt-based documentation generation for:
       - sphinx (old documentation)
       - paradox (new documentation)
       - jekyll (overview website with links to all of documentation)
       - api-v1 typescript
       - api-v2 typescript
       - api-admin swagger
- [x] add sbt-based publishing of documentation to the `gh-pages` branch for http://docs.knora.org
- [x] move `knora.org` site to https://github.com/dhlab-basel/knora.org
- [x] reimplement `knora.org` website using Jekyll (https://github.com/dhlab-basel/knora.org/pull/3)
- [x] add links to `knora.org` for `docs.knora.org` (https://github.com/dhlab-basel/knora.org/issues/2)

### To Test

- inside the `docs` folder run `sbt previewSite`. This should build and open the content for `docs.knora.org` locally.

### Issues

- resolves #733
- resolves #734

### CODE REVIEW

**Comment by subotic** [comment]
@benjamingeer whenever you have a bit time :-)

**Comment by benjamingeer** [comment]
Got this error:

```
[error] java.io.IOException: Cannot run program "jekyll" (in directory "/Users/benjamingeer/git/Knora/docs/src/jekyll"): error=2, No such file or directory
```

**Comment by benjamingeer** [comment]
Did this:

```
sudo gem install jekyll-feed
```

Now it works. :)

File: docs/Readme.md [hunk 1]
```diff
@@ -1,140 +1,115 @@
...
-### 5.2 Building the Documentation
+### Installation
 
-Inside the docs folder type for the html documentation type
+Jekyll is used for building the *Knora Documentation Overview* website, which source can be found under `src/jekyll`.
+To install the necessary dependencies, run the following commands from inside the `src/jekyll` folder:
 
 ```
-$ make jsonformat
+$ gem install jekyll bundler
```
Comment (benjamingeer) [inline docs/Readme.md]: Please add `jekyll-feed` here.

File: docs/Readme.md [hunk 2]
```diff
@@ -1,140 +1,115 @@
...
+ - `src/jekyll`: The Knora Documentation Overview Website
+ - `src/paradox`: The Paradox-Mardown-based newer style Knora general documentation
+ - `src/sphinx`: The Sphinx-based older style Knora general documentation
+ - `src/api-v1`: The Knora JSON API V1 Request and Response Format documentation. Source can be found in `salsah1/src/typescript_interfaces`
+ - `src/api-v2`: The Knora JSON-LD API V2 Request and Response Format documentation.
+ - `src/api-admin`: The Knora JSON Admin API Request and Response format Swagger-based documentation.
 
-### 2.1 Globally ###
+We are currently in a transitioning phase, where we slowly move chapters from Sphinx to Paradox. The process of 
+moving chapter involves the following Pandoc command for converting `rst` documents to `md` documents:
```
Comment (benjamingeer) [inline docs/Readme.md]: How good is the conversion? Could we convert everything all at once using a script? Or do the converted files require manual fixes?

**Comment by benjamingeer** [comment]
The new-style docs look great!

File: docs/Readme.md (hunk 2 above)
Comment (subotic) [reply docs/Readme.md]: The links between documents is an extension of markdown through paradox. So those would need to be done manually. Also, we have some graphs embedded (via `dot`), which would need to be created manually and added to markdown manually. We could certainly try, and then see what happens.

File: docs/Readme.md (hunk 1 above)
Comment (subotic) [reply docs/Readme.md]: did you run `bundle` from the `src/jekyll` directory? I think that this installs `jekyll-feed`

Comment (benjamingeer) [reply docs/Readme.md]: No, I just did what it says in the description:

> inside the docs folder run sbt previewSite

I only realised later that there were more instructions in the README. :)

File: docs/Readme.md (hunk 2 above)
Comment (benjamingeer) [reply docs/Readme.md]: OK, let's merge this first.

**Comment by subotic** [comment]
thanks :-)

I have added some more folders inside `src/jekyll` to `.gitignore`

**Comment by subotic** [comment]
Thanks for the review :-)

**Pull request merged**